### PR TITLE
[9.0][MIG] web_translate_dialog

### PR DESCRIPTION
**Pull request by antespi**
- Make it compliance to OCA guidelines
- Add Translate action if any translation field in the form
- When click in field translate icon, show modal with this field only (not all fields in form, use More dropdown action to translate all fields at once)

### CODE REVIEW

**Comment by rafaelbn** [comment]
Go to setting -> Load a translation and you will get this error:

```
Odoo Client Error

TypeError: this.sidebar is undefined

http://3150864-387-5cb43f.runbot1.odoo-community.org/web_translate_dialog/static/src/js/web_translate_dialog.js:17
Traceback:
openerp.web_translate_dialog/<.render_sidebar@http://3150864-387-5cb43f.runbot1.odoo-community.org/web_translate_dialog/static/src/js/web_translate_dialog.js:17:13
OdooClass.extend/Class.include/</prototype[name]</<@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/src/js/framework/class.js:122:35
ViewManager<.render_view_control_elements@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/src/js/view_manager.js:305:13
ViewManager<._display_view@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/src/js/view_manager.js:183:37
ViewManager<.switch_mode/switched<@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/src/js/view_manager.js:162:20
.Deferred/promise.then/</</<@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3276:86
jQuery.Callbacks/fire@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3119:38
jQuery.Callbacks/self.fireWith@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3231:49
.when/updateFunc/<@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3360:49
jQuery.Callbacks/fire@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3119:38
jQuery.Callbacks/self.fireWith@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3231:49
.when/updateFunc/<@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3360:49
jQuery.Callbacks/fire@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3119:38
jQuery.Callbacks/self.fireWith@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3231:49
.Deferred/</deferred[tuple[0]]@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3321:33
jQuery.Callbacks/fire@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3119:38
jQuery.Callbacks/self.fireWith@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3231:49
.Deferred/</deferred[tuple[0]]@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3321:33
jQuery.Callbacks/fire@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3119:38
jQuery.Callbacks/self.add@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3165:49
.Deferred/promise.then/</</<@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3278:73
jQuery.Callbacks/fire@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3119:38
jQuery.Callbacks/self.fireWith@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3231:49
.Deferred/</deferred[tuple[0]]@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3321:33
ParentedMixin.alive/</<@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/src/js/framework/mixins.js:79:21
.Deferred/promise.then/</</<@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3276:86
jQuery.Callbacks/fire@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3119:38
jQuery.Callbacks/self.fireWith@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3231:49
.Deferred/promise.then/</</<@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3283:73
jQuery.Callbacks/fire@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3119:38
jQuery.Callbacks/self.fireWith@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3231:49
.Deferred/</deferred[tuple[0]]@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3321:33
jQuery.Callbacks/fire@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3119:38
jQuery.Callbacks/self.fireWith@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3231:49
.Deferred/promise.then/</</<@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3283:73
jQuery.Callbacks/fire@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3119:38
jQuery.Callbacks/self.fireWith@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3231:49
.Deferred/promise.then/</</<@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3283:73
jQuery.Callbacks/fire@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3119:38
jQuery.Callbacks/self.fireWith@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:3231:49
done@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:9275:33
.send/callback@http://3150864-387-5cb43f.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:9685:57
```

**Comment by antespi** [comment]
@rafaelbn Error fixed, please try again

**Comment by rafaelbn** [comment]
Tested in runbot functionally :+1: , thanks!

File: web_translate_dialog/static/src/js/web_translate_dialog.js [hunk 1]
```diff
@@ -43,7 +51,19 @@ openerp.web_translate_dialog = function (instance) {
             this.view_type = parent.fields_view.type || '';
             this.$view_form = null;
             this.$sidebar_form = null;
-            this.translatable_fields_keys = _.map(this.view.translatable_fields || [], function(i) { return i.name;});
+            if (typeof(field) !== 'undefined') {
```
Comment (kutyel) [inline web_translate_dialog/static/src/js/web_translate_dialog.js]: This line can be rewritten as: 

``` javascript
if (!!field) {
```

It's an old JavaScript trick, implicitly casts to boolean as you are looking for a **truthy** value. :wink:

Comment (antespi) [inline web_translate_dialog/static/src/js/web_translate_dialog.js]: Fixed, thanks

**Comment by antespi** [comment]
Migrated to new JS API, please review it again in runbot. Thanks

**Comment by antespi** [comment]
Running as expected, please mark it as needsreview

**Comment by pedrobaeza** [comment]
Please squash all the commits from the OCA bot

File: web_translate_dialog/README.rst [hunk 2]
```diff
@@ -1,35 +1,69 @@
...
 * Autosize the textareas to the size of the content
 
 Usage
 =====
 
-Go to an object that has translatable fields (*Products* for example) and select *More > Translate* (or click on *Edit* and then click on the flag at the top-right of one of the translatable fields): the translation view will pop-up on your screen. This translation view contains all the translatable fields of the object.
+Go to an object that has translatable fields (*Products* for example) and
+select *More > Translate* (or click on *Edit* and then click on the flag at
+the top-right of one of the translatable fields): the translation view will
+pop-up on your screen. This translation view contains all the translatable
```
Comment (pedrobaeza) [inline web_translate_dialog/README.rst]: Change this to indicate that clicking on the flag, only the clicked field will be presented.

**Comment by pedrobaeza** [comment]
Little comments, but :+1:

**Comment by antespi** [comment]
Fixed and squashed

**Comment by antespi** [comment]
@lasley thanks to your sample here https://github.com/OCA/web/pull/387  I've migrated to new JS API this add-on too :smile:

**Comment by pedrobaeza** [comment]
I still see 7 comments from OCA bot

**Comment by antespi** [comment]
OCA bot squashed

**Comment by pedrobaeza** [comment]
:+1:

**Comment by pedrobaeza** [comment]
@antespi, I have found an error. Steps to reproduce:
- Activate developer mode.
- Go to Settings > Technical > Email > Templates.
- Open and edit a template.
- Click on the flag for the body field (the big one with HTML).
- You'll get a JS error.

**Comment by antespi** [comment]
@pedrobaeza You are right!. I didn't check HTML fields and in v9 UI editor for HTML fields is summernote (instead of cleditor). I've migrate JS code from cleditor to summernote library. You can check it now. Thanks.

**Comment by pedrobaeza** [comment]
Now working as expected, thanks :+1:

**Comment by rafaelbn** [comment]
Tested in runbot, I cannot find any error. Thanks :+1:

**Pull request merged**